sim/src/trips.rs

```diff
@@ -1046,6 +1090,18 @@ impl TripManager {
                 let req = maybe_req.unwrap();
                 if let Some(router) = maybe_path.map(|path| goal.make_router(vehicle.id, path, map))
                 {
+                    if !cap.allow_trip(vehicle.id, router.get_path()) {
```

Comment:
The code duplication in `TripManager` in general is horrendous. This module needs some serious rethinking.